### ISSUE TITLE
Get updated contacts from date

### ISSUE THREAD

**Issue by nagesh314** [jamesiarmes/php-ews#326]
How to get updated contacts list  ? I know full sync but is there a way for getting only updated OR deleted contacts from a particular date ? please help. 

One option is full sync and compare changekey but I have some 800 contacts and its going to be time taking oen if I do full sync every time when user logged in.

**Comment by Garethp** [comment]
If you're using the `SyncFolderItems` operation, you can store your `SyncState` to keep track of where you are in the sync. If you get the `SyncState` from your operation, store it, the next time you use it you're only fetching changes since the last sync.

**Comment by nagesh314** [comment]
Oh ok I got it Thank you so much Sir. I am storing syncstate.

**Comment by nagesh314** [comment]
Hi,

I have one more query. I am using exchange server 2007 and if I want to
know deleted calendar events then in that case I see only one option that
is full sync every time and comparing missing event ids with my database
table. Is there any other way to find deleted items so that I can delete
them from my database.

Regards,
Nagesh

**Comment by Garethp** [comment]
Why do you think you have to do a full sync every time? According to the [MSDN docs on SyncFolderItems](https://msdn.microsoft.com/en-us/library/office/aa563967(v=exchg.150).aspx), you can pass a sync state in Exchange 2007. When you do this, you don't receive all of the items as they exist at that point, you receive the *events*. So you should get DeleteItem events as well, and that'll tell you when a calendar item was deleted

**Comment by nagesh314** [comment]
yes I tried that as well but it doesn't show deleted item. that is the
reason I asked you. it only shows created items and updated items.

I deleted one event from calendar but it is not showing. yes I am using
sync state.

**Comment by Garethp** [comment]
When you do your sync state where the delete item is meant to show up, can do you a dump on the response you receive and past it to a gist or something? If you could do a dump earlier in the NTLM to get it in the raw XML, before it becomes objects, that would be best.

As a side note, are you using this jamesaires library, or did you check out my one? If you swap to my one, I might be able to give more in depth help. And is there a specific reason you're using Exchange 2007?

**Comment by nagesh314** [comment]
Hi,

I tried a lot and after a few try I started getting deleted item as well. I
was using old sync state and because of that I wasn't getting deleted item.
Now I am getting it thanks a lot for your HELP and quick response. I will
send new queries to you if any.

Regards,
Nagesh

**Comment by nagesh314** [comment]
Hi,

Auto discover is not working for me. I don't know why but here is the error
message.

*Warning*: dns_get_record(): Dns Query failed in
*C:\wamp\www\exchange\php-ews-master\EWSAutodiscover.php* on line *499*

*Warning*: Invalid argument supplied for foreach() in
*C:\wamp\www\exchange\php-ews-master\EWSAutodiscover.php* on line *361*

Regards,
Nagesh

**Comment by Garethp** [comment]
I would suggest you not use Autodiscover. You should already know the host and version you're working with

**Comment by nagesh314** [comment]
yes but my client wants autodiscover so that user dont have to remember his
server name. Please suggest.

**Comment by nagesh314** [comment]
I tried first example from this url but error

https://github.com/jamesiarmes/php-ews/wiki/Autodiscovery

**Comment by nagesh314** [comment]
Hi,

if I use $server = 'outlook.office365.com'; or $server = 'outlook.com'; it
works well for me but I have question

will it work for all exchange server versions? or should I use first one
for autodiscover.

Regards,
Nagesh

**Comment by Garethp** [comment]
I suggest not using Autodiscover, because it can take three or four requests before finding the server. Or it could make 5 requests and not find it at all. It can slow down the application a lot of you're doing Autodiscovers all the time. 

Also, that won't work if your exchange server isn't hosted online. If it's a locally hosted server, by a different name, then that won't work

**Comment by nagesh314** [comment]
Autodiscover will get used only once after that based on response I will
store server name in database. But the problem is its not working for me.

**Comment by Garethp** [comment]
What's the email address that you're trying to use to autodiscover?

**Comment by nagesh314** [comment]
Hi,

Its working on live server but not on wamp server.

Regards,
Nagesh

**Comment by Garethp** [comment]
What's the email address that you're trying to use to autodiscover?

**Comment by nagesh314** [comment]
I asked client he said please do not share email address this is clients
email address.

**Comment by Garethp** [comment]
Well, can you post it with just the domain? I can't do much to help you, but I can do even less if I don't know what domain you're trying to query

**Comment by nagesh314** [comment]
Hi,

How to split threaded email messages in PHP EWS. I am not getting any
solution online.

For example. I want to read only single message at a time and not entire
thread.

Regards,
Nagesh

**Comment by nagesh314** [comment]
Hi,

Are you there please reply to my email. I am confused with exchange email
thread.

Regards,
Nagesh

**Comment by Garethp** [comment]
The messages are already split. You should only be getting one at a time

**Comment by nagesh314** [comment]
oh is it. but when there is conversation between 2 users and when they
exchange emails with same subject line message id remains same. let me me
paste example here. Now in the following email message there are 2 text
body which I want separate for inserting it in mysql with sender email id
etc.

*1. good to see this message. *
*2. this is the test by exchange mail id*

-------------------------------------------------------------------------------
good to see this message.

On Wed, Jun 1, 2016 at 6:57 PM, Nagesh Sawant <nsawant@dllbuilder.com>
wrote:
this is the test by exchange mail id
----------------------------------------------------------------------------

**Comment by nagesh314** [comment]
when I pass message id it is showing body text as

["Body"]=>
object(stdClass)#17 (2) {
["_"]=>
string(438) "<div dir="ltr">good to see this message.Â </div><div class="
gmail_extra"><br><div class="gmail_quote">On Wed, Jun 1, 2016 at 6:57 PM,
Nagesh Sawant <span dir="ltr">&lt;<a href="mailto:nsawant@dllbuilder.com"
target="_blank">nsawant@dllbuilder.com</a>&gt;</span> wrote:<br><blockquote
class="gmail_quote" style="margin:0 0 0 .8ex;border-left:1px #ccc
solid;padding-left:1ex">this is the test by exchange mail id</blockquote>
</div><br></div>
"
["BodyType"]=>
string(4) "HTML"
}

**Comment by Garethp** [comment]
Yeah, you'll have to come up with a way to do that yourself

**Comment by nagesh314** [comment]
syncfolderitem will return only 512 results what if there are more than 512
results? will it next time return other 512 results or is there any other
work around paging type of.

$request = new EWSType_SyncFolderItemsType;
$request->SyncState = $sync_state;
$request->MaxChangesReturned = 512;

**Comment by Garethp** [comment]
It will return the **next** 512 items *from* the SyncState that you passed in. So it's automatically paged

**Comment by nagesh314** [comment]
ok Thank you so much.

**Comment by nagesh314** [comment]
Hi,

Autodiscover on live server works fine but there is is error for contact
get list or for sync folder request. What can be the issue.

*Fatal error*: Uncaught SoapFault exception: [soap:Client] The specified
server version is invalid. in

in */home/public_html/application/libraries/EWS/ExchangeWebServices.php* on
line *391*

Regards,
Nagesh

**Comment by Garethp** [comment]
The issue could be that you specified the wrong server version for your server or feature?

**Comment by nagesh314** [comment]
Hi,

I am using email inbox sync but its getting timeout because there are too
many emails are there in the inbox and I am using while loop.

//IncludesLastItemInRange
while($lastItemInRange != 1) {

what to do in this case please suggest.

Regards,
Nagesh

**Comment by Garethp** [comment]
Don't use a loop. If you're storing and re-using the sync state, then you can sync between 100-1000 items per script. Meaning instead of Running it in a loop, you can just schedule the script to run once a minute, or once every 5 minutes, and then it'll catch up on it's own

**Comment by nagesh314** [comment]
Hi,

Do you know which is the best solution for implementing video chat in
website.

Regards,
Nagesh

**Comment by nagesh314** [comment]
Hi,

I have one question. I am using EWS authentication but how to store user
password. Oauth2 is only available in exchange 365 server.

What is the option for storing passord in database.

Regards,
Nagesh

**Comment by nagesh314** [comment]
Hi,

When I use login details of exchange server 2010 it gives me error for eg.
my autodiscover doesnt work. contact list page gives error like this

*Fatal error*: Class 'EWS_Exception' not found in
*C:\wamp\www\exchange\php-ews-master\ExchangeWebServices.php* on line *1221*

*what can be the issue My curl version is also > 7.23*

**Comment by nagesh314** [comment]
Hi,

When I use login details of exchange server 2010 it gives me error for eg.
my autodiscover doesnt work. contact list page gives error like this

*Fatal error*: Class 'EWS_Exception' not found in
*C:\wamp\www\exchange\php-ews-master\ExchangeWebServices.php* on line *1221*

*what can be the issue My curl version is also > 7.23*

**Comment by nagesh314** [comment]
*Fatal error*: Uncaught exception 'EWS_Exception' with message 'SOAP client
returned status of 401' in
C:\wamp\www\exchange\php-ews-master\ExchangeWebServices.php:1221 Stack
trace: #0 C:\wamp\www\exchange\php-ews-master\ExchangeWebServices.php(547):
ExchangeWebServices->processResponse(NULL) #1
C:\wamp\www\exchange\contact_list.php(58):
ExchangeWebServices->FindItem(Object(EWSType_FindItemType)) #2 {main}
thrown in *C:\wamp\www\exchange\php-ews-master\ExchangeWebServices.php* on
line *1221*

**Comment by nagesh314** [comment]
for exchange 2010 I still get error please provide some solution..

*Fatal error*: Uncaught exception 'EWS_Exception' with message 'SOAP client
returned status of 401' in
C:\wamp\www\exchange\php-ews-master\ExchangeWebServices.php:1221 Stack
trace: #0
C:\wamp\www\exchange\php-ews-master\ExchangeWebServices.php(1064):
ExchangeWebServices->processResponse(NULL) #1
C:\wamp\www\exchange\contact_sync.php(63):
ExchangeWebServices->SyncFolderItems(Object(EWSType_SyncFolderItemsType))
#2 {main} thrown in
*C:\wamp\www\exchange\php-ews-master\ExchangeWebServices.php* on line *1221*

**Comment by nagesh314** [comment]
https://github.com/Garethp/php-ews/issues/65

**Comment by jamesiarmes** [comment]
@nagesh314 I'm having trouble following this thread as there seem to be multiple issues that you're looking for support with. As it seems your original issue has been resolved, I am closing this issue. If you have further issues that you need support with, please create a new issue for each individually.
